packages/url_launcher/ios/Classes/UrlLauncherPlugin.m

```diff
@@ -165,4 +149,35 @@ - (void)closeWebViewWithResult:(FlutterResult)result API_AVAILABLE(ios(9.0)) {
   result(nil);
 }
 
+- (UIViewController *)topViewController {
+  return [self topViewControllerFromViewController:[UIApplication sharedApplication].keyWindow.rootViewController];
+}
+
+/** 
+ * This method recursively iterate through the view hierarchy
+ * to return the top most view controller.
+ *
+ * It supports the following scenarios:
+ *
+ * - The view controller is presenting another view.
+ * - The view controller is a UINavigationController.
+ * - The view controller is a UITabBarController.
+ *
+ * @return The top most view controller.
+ */
+- (UIViewController *)topViewControllerFromViewController:(UIViewController *)rootViewController
```

Comment:
nits: here the name `rootViewController` doesn't make sense anymore since it is a recursive method. Maybe just `viewController`?